ext/spl/spl_fixedarray.c

```diff
@@ -195,23 +195,18 @@ static HashTable* spl_fixedarray_object_get_gc(zend_object *obj, zval **table, i
 	return ht;
 }
 
-static HashTable* spl_fixedarray_object_get_properties(zend_object *obj)
+static HashTable* spl_fixedarray_get_properties_for(zend_object *obj, zend_prop_purpose purpose)
 {
 	spl_fixedarray_object *intern = spl_fixed_array_from_obj(obj);
-	HashTable *ht = zend_std_get_properties(obj);
 
-	if (!spl_fixedarray_empty(&intern->array)) {
-		zend_long j = zend_hash_num_elements(ht);
+	/* Keep the values and properties separate*/
+	HashTable *ht = zend_array_dup(zend_std_get_properties(obj));
```

Comment:
https://github.com/php/php-src/pull/8046

> In order for a native datastructure to correctly implement
> `*get_properties_for` for `var_export`'s cycle detection,
> it would need to return the exact same array every time prior to this PR. (#8046)

I was working with `get_properties_for` again after the last review, I'm pretty sure this will cause new problems with var_export on `get_properties_for` if #8046 isn't merged, for

```php
$arr = new SplFixedArray(1);
$arr[0] = $arr;
var_export($arr);  // now calls get_properties_for with purpose *VAR_EXPORT
```